### PR TITLE
Add feature #27: Map binary value in dataset and relevant unit testing.

### PR DESCRIPTION
**Pull request by soisik**
Add feature #27 to map binary value in dataset (such as images)
- Added 2 new API methods getDatasetFirstBinaryValueById(attribute) and getDatasetBinaryValuesById(attribute) to be able to access byte[] values;
- Impacted unit tests

### CODE REVIEW

File: src/main/java/org/lsc/utils/RhinoJScriptEvaluator.java [hunk 1]
```diff
@@ -94,7 +94,8 @@ public RhinoJScriptEvaluator(boolean debug) {
     }
 
     /** {@inheritDoc} */
-    public String evalToString(final Task task, final String expression, final Map<String, Object> params)
+    @Override
+	public String evalToString(final Task task, final String expression, final Map<String, Object> params)
```
Comment (rouazana) [inline src/main/java/org/lsc/utils/RhinoJScriptEvaluator.java]: indentation issue?

File: src/main/java/org/lsc/utils/RhinoJScriptEvaluator.java [hunk 2]
```diff
@@ -110,47 +111,45 @@ public String evalToString(final Task task, final String expression, final Map<S
     }
 
     /** {@inheritDoc} */
-    public List<String> evalToStringList(final Task task, final String expression, final Map<String, Object> params)
+	@Override
+	public List<Object> evalToObjectList(final Task task, final String expression, final Map<String, Object> params)
```
Comment (rouazana) [inline src/main/java/org/lsc/utils/RhinoJScriptEvaluator.java]: here also?

File: src/main/java/org/lsc/utils/RhinoJScriptEvaluator.java [hunk 3]
```diff
@@ -110,47 +111,45 @@ public String evalToString(final Task task, final String expression, final Map<S
...
         try {
             result = convertJsToJava(instanceEval(task, expression, params));
         } catch (EvaluatorException e) {
             throw new LscServiceException(e);
         }
-
+		if (result == null) {
+			return null;
+		}
+		List<Object> resultsArray = new ArrayList<Object>();
```
Comment (rouazana) [inline src/main/java/org/lsc/utils/RhinoJScriptEvaluator.java]: idem?

**Comment by soisik** [comment]
- Sorry for the added annotations, Eclipse was adding them automatically on save actions. I turned it off.
- I've corrected the ident issues on RhinoJScriptEvaluator (same problem,  autoformat on save action overridding the configuration to use spaces that  render better in git diffs)

File: src/main/java/org/lsc/utils/ScriptableEvaluator.java [hunk 4]
```diff
@@ -27,7 +27,7 @@ public String evalToString(final Task task, final String expression, final Map<S
      * @return the evaluation result, null if nothing
      * @throws LscServiceException thrown when a technical error is encountere
      */
-    public List<String> evalToStringList(final Task task, final String expression, final Map<String, Object> params)
+	public List<Object> evalToObjectList(final Task task, final String expression, final Map<String, Object> params)
```
Comment (rouazana) [inline src/main/java/org/lsc/utils/ScriptableEvaluator.java]: still an indent issue?

File: src/main/java/org/lsc/utils/RhinoJScriptEvaluator.java [hunk 5]
```diff
@@ -110,46 +111,44 @@ public String evalToString(final Task task, final String expression, final Map<S
...
+        } else if (result.getClass().isArray() && result.getClass().getComponentType().equals(byte.class)) {
+            resultsArray.add(result);
         } else {
-            List<String> resultsArray = new ArrayList<String>();
             if (result != null) {
                 resultsArray.add(result.toString());
             }
-            return resultsArray;
         }
+		return resultsArray;
```
Comment (rouazana) [inline src/main/java/org/lsc/utils/RhinoJScriptEvaluator.java]: here also?

**Pull request merged**